### PR TITLE
refactor(ex_docmd.c): resolve most clint errors

### PR DESCRIPTION
**Pull request by lewis6991**

### CODE REVIEW

File: src/nvim/ex_docmd.c [hunk 1]
```diff
@@ -721,15 +687,13 @@ int do_cmdline(char *cmdline, LineGetter fgetline, void *cookie, int flags)
...
+  // Continue executing command lines when:
+  // - no CTRL-C typed, no aborting error, no exception thrown or try
+  //   conditionals need to be checked for executing finally clauses or
+  //   catching an interrupt exception
+  // - didn't get an error message or lines are not typed
+  // - there is a command after '|', inside a :if, :while, :for or :try, or
+  //   looping for ":source" command or function call.
   while (!((got_int || (did_emsg && force_abort) || current_exception)
            && cstack.cs_trylevel == 0)
          && !(did_emsg
```
Comment (zeertzjq) [inline src/nvim/ex_docmd.c:699]: Change the `;` below to `{}`

Comment (lewis6991) [reply src/nvim/ex_docmd.c:699]: I tried that but it didn't work. It seemed like the expression was too complex for the linter or something.

Comment (zeertzjq) [reply src/nvim/ex_docmd.c:699]: Oh, I didn't notice that this is a `do` ... `while`. Then I know how to fix this.

File: src/nvim/ex_docmd.c [hunk 2]
```diff
@@ -984,20 +931,17 @@ static void store_loop_line(garray_T *gap, char *line)
   p->lnum = SOURCING_LNUM;
 }
 
-/// If "fgetline" is get_loop_line(), return TRUE if the getline it uses equals
-/// "func".  * Otherwise return TRUE when "fgetline" equals "func".
+/// If "fgetline" is get_loop_line(), return true if the getline it uses equals
+/// "func".  * Otherwise return true when "fgetline" equals "func".
 ///
 /// @param cookie  argument for fgetline()
 int getline_equal(LineGetter fgetline, void *cookie, LineGetter func)
```
Comment (zeertzjq) [inline src/nvim/ex_docmd.c:938]:
```suggestion
bool getline_equal(LineGetter fgetline, void *cookie, LineGetter func)
```

File: src/nvim/ex_docmd.c [hunk 3]
```diff
@@ -3407,13 +3322,9 @@ const char *set_one_cmd_context(expand_T *xp, const char *buff)
...
-      }
-      /* An argument can contain just about everything, except
-       * characters that end the command and white space. */
-      else if (c == '|'
-               || c == '\n'
-               || c == '"'
-               || ascii_iswhite(c)) {
+      } else if (c == '|' || c == '\n' || c == '"' || ascii_iswhite(c)) {
+        // An argument can contain just about everything, except
+        // characters that end the command and white space. */
```
Comment (zeertzjq) [inline src/nvim/ex_docmd.c:3327]:
```suggestion
        // An argument can contain just about everything, except
        // characters that end the command and white space.
      } else if (c == '|' || c == '\n' || c == '"' || ascii_iswhite(c)) {
```

File: src/nvim/ex_docmd.c [hunk 4]
```diff
@@ -138,9 +136,7 @@ struct dbg_stuff {
 # define ex_language            ex_ni
 #endif
 
-/*
- * Declare cmdnames[].
- */
+/// Declare cmdnames[].
```
Comment (zeertzjq) [inline src/nvim/ex_docmd.c:139]: I don't think a doxygen comment does anything here.
```suggestion
// Declare cmdnames[].
```

**Review by zeertzjq — APPROVED** [review]

**Pull request merged**